node/actors/executor/src/attestation.rs

```diff
@@ -1,201 +1,35 @@
-                let batch = AttesterRunner::wait_for_batch_to_sign(
-                    ctx,
-                    batch_number,
-                    &self.batch_store,
-                    self.poll_interval,
-                )
-                .await?;
-
-                // The certificates might be collected out of order because of how gossip works;
-                // we could query the DB to see if we already have a QC, or we can just go ahead
-                // and publish our vote, and let others ignore it.
-
-                tracing::info!(%batch_number, "publishing attestation");
-
-                // We only have to publish a vote once; future peers can pull it from the register.
-                self.publisher
-                    .publish(attesters, &genesis, &self.attester.key, batch)
-                    .await
-                    .context("publish")?;
-
-                ctx::Ok(())
-            }
-            .instrument(tracing::info_span!("attestation_iter"))
-            .await?;
-        }
-    }
-
-    /// Wait for the batch commitment to become available.
-    #[tracing::instrument(skip_all, fields(l1_batch = %number))]
-    async fn wait_for_batch_to_sign(
-        ctx: &ctx::Ctx,
-        number: attester::BatchNumber,
-        batch_store: &BatchStore,
-        poll_interval: time::Duration,
-    ) -> ctx::Result<attester::Batch> {
-        loop {
-            if let Some(batch) = batch_store
-                .batch_to_sign(ctx, number)
-                .await
-                .context("batch_to_sign")?
-            {
-                return Ok(batch);
-            } else {
-                ctx.sleep(poll_interval).await?;
-            }
-        }
-    }
-}
+use zksync_concurrency::{ctx, time};
+pub use zksync_consensus_network::gossip::attestation::*;
 
 /// An interface which is used by attesters and nodes collecting votes over gossip to determine
 /// which is the next batch they are all supposed to be voting on, according to the main node.
 ///
 /// This is a convenience interface to be used with the [AttestationStatusRunner].
 #[async_trait::async_trait]
-pub trait AttestationStatusClient: 'static + Send + Sync {
+pub trait Client: 'static + Send + Sync {
     /// Get the next batch number for which the main node expects a batch QC to be formed.
     ///
     /// The API might return an error while genesis is being created, which we represent with `None`
     /// here and mean that we'll have to try again later.
-    ///
-    /// The genesis hash is returned along with the new batch number to facilitate detecting reorgs
-    /// on the main node as soon as possible and prevent inconsistent state from entering the system.
-    async fn attestation_status(
-        &self,
-        ctx: &ctx::Ctx,
-    ) -> ctx::Result<Option<(attester::GenesisHash, attester::BatchNumber)>>;
+    async fn config(&self, ctx: &ctx::Ctx) -> ctx::Result<Option<Config>>;
```

Comment:
I'm not convinced that the main node API should return the committee as part of the `Config` and the attesters should just take it, as opposed to executing the batch and deriving the attester committee from their VM. I might be misunderstanding what the `Client` is supposed to be, but the docs still say it's querying the main node.

I mean you could argue that the ultimate authority about the committee is the L1 contract where the signatures are going to be checked, so even if the main node would try to get it signed by a bogus committee it would be rejected, but it seems odd that we ask the main node for something that we (will) have computed locally.

If this is a temporary measure it's not well documented. I understand we have Proof-of-Authority and centrally set the weights, and that we don't have the integration with the smart contracts yet, so the information has to come from somewhere, but maybe there is a way to show this in the APIs and data paths without bundling up the status and the configuration.

Reply:
I see the `Client` is gone, so this conversation is moot.

Reply:
ptal at how the new api will be used in practice: https://github.com/matter-labs/zksync-era/pull/2657